### PR TITLE
Added tests for EncryptionProperty (coverage 100%)

### PR DESCRIPTION
**Pull request by StanislavUshakov**
Overall code coverage for System.Security.Cryptography.Xml is 71.1%

Part of #16829

### CODE REVIEW

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            XmlDocument doc = new XmlDocument();
+            doc.LoadXml(validXml);
+            EncryptionProperty encryptionProperty = new EncryptionProperty(doc.DocumentElement);
+            Assert.NotNull(encryptionProperty.PropertyElement);
+        }
+
+        [Fact]
+        public void ConstructorWithXmlElement_Element_Null()
+        {
+            Assert.Throws<ArgumentNullException>(() => new EncryptionProperty(null));
```
Comment (hughbe) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: Nit: for this or in the future, include the param name for exceptions for DerivedException: 
```
Assert.Throws<ArgumentNullException>("...", () => new EncryptionProperty(null));
```

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            Assert.Null(encryptionProperty.PropertyElement);
+        }
+
+        [Fact]
+        public void ConstructorWithXmlElement_Valid()
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.LoadXml(validXml);
+            EncryptionProperty encryptionProperty = new EncryptionProperty(doc.DocumentElement);
+            Assert.NotNull(encryptionProperty.PropertyElement);
```
Comment (hughbe) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: Can we get a bit more info from this assertion? It's not just about covering the code path but also making sure it does the right thing

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class EncryptionPropertyTest
+    {
+        private const string validXml = @"<EncryptionProperty xmlns=""http://www.w3.org/2001/04/xmlenc#"" Id=""ID_val"" Target=""Target_val"" />";
```
Comment (hughbe) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: Nit: validXml => ValidXml

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        }
+
+        [Fact]
+        public void ConstructorWithXmlElement_Element_Null()
+        {
+            Assert.Throws<ArgumentNullException>(() => new EncryptionProperty(null));
+        }
+
+        [Fact]
+        public void ConstructorWithXmlElement_InvalidXml()
```
Comment (hughbe) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: Nit: can we get a bit more information on why this is valid in the name
```
public void Ctor_VoidXmlElement_ThrowsCryptographicException()
```

Not sure if that's the right name, but you get the idea

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        {
+            string xml = "<a />";
+            XmlDocument doc = new XmlDocument();
+            doc.LoadXml(xml);
+            Assert.Throws<CryptographicException>(() => new EncryptionProperty(doc.DocumentElement));
+        }
+
+        [Fact]
+        public void SetPropertyElement_Null()
```
Comment (hughbe) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: I think across corefx we go for a pattern of naming tests along the lines of

MethodOrPropertyTested_SpecificScenario_ExpectedOutcome

e.g. here
```cs
public void PropertyElement_SetNull_ThrowsArgumentNullException()
```

It's not a law and takes some getting used to, don't feel like you have to go around changing existing tests or tests in this PR :)

(Oh and great work with these tests)

**Comment by krwq** [comment]
Thank you @StanislavUshakov, good work! Overall LGTM, the only thing I'd like to have fixed in this PR is @hughbe 's comment about more testing around NotNull (ConstructorWithXmlElement_Valid) everything else is optional but a good thing to have

**Comment by StanislavUshakov** [comment]
Thank you for the rview, will fix it tomorrow.

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        [Fact]
+        public void LoadXml_ValidEncryptionPropertyXml()
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.LoadXml(ValidXml);
+            EncryptionProperty encryptionProperty = new EncryptionProperty();
+            encryptionProperty.LoadXml(doc.DocumentElement);
+            Assert.Equal(doc.DocumentElement, encryptionProperty.PropertyElement);
+            Assert.Equal("ID_val", encryptionProperty.Id);
+            Assert.Equal("Target_val", encryptionProperty.Target);
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: These strings probably make more sense as consts up near the XML.  I was confused as to if they were from data (they were) or a spec, or were defaults, or what.

**Comment by StanislavUshakov** [comment]
@dotnet-bot test Innerloop OSX10.12 Debug x64 Build and Test

File: src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs (hunk 6 above)
Comment (StanislavUshakov) [inline src/System.Security.Cryptography.Xml/tests/EncryptionPropertyTest.cs]: Yes, didn't move them to const after moving xml to const, sorry

**Comment by krwq** [comment]
OSX is still having networking issues, I think it is ok to merge

**Pull request merged**